src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,42 @@
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "shared/constants.h"
+
+/**
+ * This struct represents the circular buffer properties
+ */
+typedef struct CircularBuffer CircularBuffer_t;
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  cbuffer The circular_buffer
+ * @param  size Size of the circular_buffer to set
+ */
+CircularBuffer_t *circular_buffer_create(CircularBuffer_t *cbuffer, int size);
+
+/**
+ * Destroy the circular_buffer, freeing any memory allocated for it
+ *
+ * NOTE: This will not destroy the values pointed to by any pointers passed to the
+ * `create` function
+ *
+ * @param  cbuffer The circular_buffer to destroy
+ */
+void circular_buffer_destroy(CircularBuffer_t *cbuffer);
+
+/**
+ * Push data into the circular_buffer
+ * If full, will overwrite the oldest values
+ * @param  cbuffer The circular_buffer
+ * @param  data  Value to be inserted into the circular_buffer
+ */
+void circular_buffer_push(CircularBuffer_t *cbuffer, float data);
+
+/**
+ * Retrieve the most recent value in the circular_buffer
+ * Ex. 1 is the most recent value, 2 is the 2nd most recent value, etc.
+ * @param  cbuffer The circular_buffer
+ * @param  index Index of most recent value
+ */
```

Comment:
We should add a note that then index must be less then the max size

Reply:
We should also add a `@return`

Reply:
Added note. I added an `@return` but I'm not sure how to word the data returned is the most recent based on the index. Would appreciate if someone could phrase it better.

Reply:
It's also not clear what `0` returns, or what negative values return (since it seems like that's valid in the implementation)?

I'd say something like `the index is relative to the most recent value in the buffer. Eg. 1 is the most recent value, 2 is the second most recent, etc`

Reply:
@kelvinkoon - not sure if you've implemented Mat's suggested change here? the current `@return` is still a bit confusing to me.

Reply:
I added Mat's comment to the description in the header. I'll phrase it in the `@return` instead.

Reply:
Changed the `@return`, let me know if it makes more sense.